components/omega/src/base/Halo.h

```diff
@@ -132,7 +132,7 @@ class Halo {
       /// index space. 0 = OnCell, 1 = OnEdge, 2 = OnVertex
       ExchList SendLists[3], RecvLists[3];
       /// Buffers for MPI communication
-      std::vector<Real> SendBuffer, RecvBuffer;
+      std::vector<R8> SendBuffer, RecvBuffer;
```

Comment:
It is because we still might have to do halo exchanges of double precision arrays when configured to use single precision (for example, if we keep some fields in double precision). The way that packing of double precision arrays is done right now requires the buffers to be able to hold double precision values: https://github.com/E3SM-Project/Omega/blob/80182872ef256bac39040ce1e8be858949df1347/components/omega/src/base/Halo.cpp#L752-L766

This could be avoided by doing packing not at the value level, but using the byte representation. I didn't want to do that in this PR, since it is a fairly large change, and we know that halo exchanges need to be refactored anyway to avoid unnecessary transfers for GPUs.

Reply:
That all sounds good.